src/daemon/emitter.cc

```diff
@@ -579,6 +608,27 @@ void Emitter::DoRemoteExecute(const base::WorkerPool& pool, ResolveFn resolver,
     failed_tasks_->Push(std::move(*task));
     counter.ReportOnDestroy(true);
   }
+
+  auto new_conf = this->conf();
+
+  // In case if pool was shut down to apply new configuration check if
+  // new number of shards is smaller than our shard. If so, take out tasks
+  // from our shard and distribute them across other shards.
+  if (!all_tasks_->IsClosed() &&
+      new_conf->emitter().shard_queue_limit() > 0 &&
+      new_conf->emitter().total_shards() <= shard) {
+    bool shard_is_empty = false;
+    do {
+      Optional&& task =
+          all_tasks_->Pop(pool, conf->emitter().shard_queue_limit(), shard);
```

Comment:
Then the default value(`0`) will be used. This means that all tasks without shards will be popped and distributed across new shards. Looks like a valid behavoir to me.